sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -22,14 +22,67 @@
 
 from __future__ import absolute_import
 
-from apache_beam import coders
+from apache_beam.io.iobase import Read
+from apache_beam.io.iobase import Write
 from apache_beam.runners.dataflow.native_io import iobase as dataflow_io
+from apache_beam.transforms import PTransform
+from apache_beam.transforms import ParDo
 from apache_beam.transforms.display import DisplayDataItem
 
-__all__ = ['PubSubSink', 'PubSubSource']
 
+__all__ = ['ReadStringsFromPubSub', 'WriteStringsToPubSub']
 
-class PubSubSource(dataflow_io.NativeSource):
+
+class ReadStringsFromPubSub(PTransform):
```

Comment:
Can we rename this `ReadFromPubsub` and pass in a coder as a kwarg, defaulting to `coders.StrUtf8Coder()`, saving the coder to `self._coder`?  This makes the interface more in line with existing implementations like `ReadFromText` / `ReadFromAvro` / `ReadFromTfRecord`.